src/audio/google/google_rtc_audio_processing.c

```diff
@@ -618,38 +620,14 @@ static int google_rtc_audio_processing_prepare(struct processing_module *mod,
 	return 0;
 }
 
-/* Extracts connected buffer pointers.  Clumsy, should cache in comp_data */
-static void find_bufs(struct processing_module *mod, struct comp_buffer **mic,
-		      struct comp_buffer **ref, struct comp_buffer **out)
-{
-	struct google_rtc_audio_processing_comp_data *cd = module_get_private_data(mod);
-	struct list_item *li;
-	int i = 0;
-
-	list_for_item(li, &mod->dev->bsource_list) {
-		struct comp_buffer *b = container_of(li, struct comp_buffer,
-						     sink_list);
-		if (i == cd->raw_microphone_source)
-			*mic = b;
-		else if (i == cd->aec_reference_source)
-			*ref = b;
-		i++;
-	}
-
-	*out = list_first_item(&mod->dev->bsink_list,
-			       struct comp_buffer, source_list);
-}
-
 static int trigger_handler(struct processing_module *mod, int cmd)
 {
-	struct comp_buffer *mic, *ref, *out;
-
-	find_bufs(mod, &mic, &ref, &out);
+	struct google_rtc_audio_processing_comp_data *cd = module_get_private_data(mod);
 
 	/* Ignore and halt propagation if we get a trigger from the
 	 * playback pipeline: not for us.
 	 */
-	if (ref->walking)
+	if (cd->ref_comp_buffer->walking)
```

Comment:
check for NULL? A trigger before the reference pipeline is prepared?

Reply:
It's computed in prepare(), but indeed there's no error check on the pipeline not being present.  I'll add that.